tests/k6/src/lib/api/api.ts

```diff
@@ -0,0 +1,12 @@
+import encoding from 'k6/encoding';
+import * as types from "../types";
+
+export const headersDefault = ({credential}: { credential: types.Credential }): { [key: string]: string } => {
+    const isOIDCGuard = (credential as types.Token).tokenType !== undefined;
+    const authOIDC = credential as types.Token;
+    const authBasic = credential as types.Account;
+
+    return {
+        Authorization: isOIDCGuard ? `${authOIDC.tokenType} ${authOIDC.accessToken}` : `Basic ${encoding.b64encode(`${authBasic.login}:${authBasic.password}`)}`,
```

Comment:
you can probably add a function on each auth Providers to return the formatted Authorization header and use that here.

Reply:
yea but this way we bloat the auth providers with too much responsibility. In my opinion we should seperate by concerns here. The Provider has nothing to do with a header or any protocol.

its responsible for getting the information needed, for example the token or the auth credentials. 
the headers are clear bounded to the context of http (in our case) and maybe only those need a auth header.

if we implement a grpc bridge for example. They need the token but not the header.